b9lab-content/3-main-concepts/02-architecture.md

```diff
@@ -1 +1,294 @@
+
+<HighlightBox type="info">
+More detailed information the ABCI can be found here:
+
+* A prose specification [https://github.com/tendermint/abci/blob/master/specification.md]
+* A protobuf file [https://github.com/tendermint/abci/blob/master/types/types.proto]
+* A Go interface [https://github.com/tendermint/abci/blob/master/types/application.go]
+* Documentation [http://tendermint.readthedocs.io/en/master/]:
+</HighlightBox>
+
+## State Machines
+
+At its core, a blockchain is a replicated state machine. A state machine is a computer science concept, in which a machine can have multiple states but only one state at a time. And, there is a state transition process, or a set of defined processes which are the only way the state changes from the old state to a new state.
+
+![state machine](./images/state_machine_1.png)
+
+The state transition function in a blockchain is virtually synonymous with a transaction. Given an initial state, a confirmed transaction and a set of rules for interpreting that transaction, the machine transitions to a new state. The rules of interpretation are defined at the application layer.
+
+Blockchains are deterministic, so the only correct interpretation of the transaction is the new state, shown as S-prime in the illustration above.
+
+Blockchains are distributed and, in practice, transactions arrive in batches called blocks. There is a state machine that exists after the correct interpretation of each transaction in the block. Each transaction executes in the context of the state machine that resulted from every transaction that preceded it. The state machine that exists after all transactions in the block have been executed is a useful checkpoint, especially when one is interested in historic states.
+
+![state machine](./images/state_machine_2.png)
+
+Developers create the state machine using the Cosmos SDK. This includes storage organization, also known as the state, and the state transaction functions which determine what is permissible and what, if any, adjustments to the state result from each kind of transaction.
+
+In this context, “consensus” establishes a canonical set of well-ordered blocks that contain well-ordered transactions. The chain of blocks is therefore a well-ordered set of all finalized transactions and all nodes agree that the canonical set is the only relevant set. Since the state machine defined by the developers is deterministic, there is only one correct interpretation of the canonical transaction set and, by extension, only one correct determination of the state machine when any given transaction is executed or at any block height.
+
+This state machine definition is silent on the processes that confirm and propagate transactions, and Tendermint is agnostic about the meaning of the blocks it organizes.The Tendermint consensus establishes the ordered set of transactions. By extension, the nodes reach consensus about the state of the application.
+
+![ABCI](./images/ABCI_2.png)
```

Comment:
Although great, I'm wondering if this section is necessary in this already very dense chapter. If the end goal is understanding the Long-running Exercise, I'm not entirely sure if this level of abstraction is required. Perhaps this could be an expansion panel instead?